LFS-34: UI for viewing local users and editing them.

zm2he opened the pull request:
An implementation of the UI is found at `http://localhost:8080/content.userBoard.html`. Currently, this is part of the homepage module; it should be separated into its own module later. 

The current UI allows local users to be created and deleted; their passwords can also be changed

marta- committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Added media folder and possible background image

marta- committed:
[misc] Added media folder and possible background image

marta- committed:
[misc] Added media folder and possible background image

Commit:
LFS-32: Enabling Brian's LFS login portal

Commit:
LFS-32: Enabling Brian's LFS login portal

zm2he committed:
LFS-34: Enabled Brian's login form to appear at localhost:8080/content.loginSignUp.html.

zm2he committed:
LFS-34: Enabled Brian's login form to appear at localhost:8080/content.loginSignUp.html.

Commit:
LFS-34: Workaround for loginValidationPOSTPath and loginRedirectionPath variables; delete unneeded 'login' folder

Commit:
LFS-34: Workaround for loginValidationPOSTPath and loginRedirectionPath variables; delete unneeded 'login' folder

Commit:
LFS-34: Correct path for login verification added.

Commit:
LFS-34: Configured default login page to new React login page. Removes /j_security_check redirect on login failure.

Commit:
LFS-34: Configured default login page to new React login page. Removes /j_security_check redirect on login failure.

marta- committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Added dependencies to semantic-ui and material-react-dashboard

marta- committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Added dependencies to semantic-ui and material-react-dashboard

marta- committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Added dependencies to semantic-ui and material-react-dashboard

Commit:
Initial code for user management interface. Does not work currently.

Commit:
Initial code for user management interface. Does not work currently.

Commit:
LFS-34: Merge with dev for PR #12 build updates.

Commit:
LFS-34: Temporary setup for testing user account handling.

Commit:
LFS-34: Temporary setup for testing user account handling.

zm2he committed:
LFS-34: Resolved error with this.setState() when updating list of users.

zm2he committed:
LFS-34: Resolved error with this.setState() when updating list of users.

Commit:
In process of adjusting GET for user information.

Commit:
In process of adjusting GET for user information.

zm2he committed:
Enabled GET for user information from the system instead of the bin directory.

zm2he committed:
Enabled GET for user information from the system instead of the bin directory.

Commit:
Very basic functioning prototype for user management. Can list users, create user, delete user, and change user password.

Commit:
Very basic functioning prototype for user management. Can list users, create user, delete user, and change user password.

Commit:
Some experiments with dialogues.

Commit:
Some experiments with dialogues.

Commit:
ComponentWillMount now loads list of users automatically upon opening content.userBoard.html

Commit:
ComponentWillMount now loads list of users automatically upon opening content.userBoard.html

Commit:
Restored styles to userBoard.js and fixed problems with dialog experiment.

Commit:
Restored styles to userBoard.js and fixed problems with dialog experiment.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Copy material-react-dashboard Dashboard & navbar

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Add sidebar from material-dashboard-react

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Add sidebar from material-dashboard-react

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Copy material-react-dashboard Dashboard & navbar

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Copy material-react-dashboard Dashboard & navbar

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Add sidebar from material-dashboard-react

Commit:
Put login page in separate module

Commit:
Put login page in separate module

Commit:
Merged dev into LFS-51 for updates

Commit:
Deleted extraneous and redundant code from hompage module: dist/SLING-INF/content/apps and code associated with login.

Commit:
Deleted extraneous and redundant code from hompage module: dist/SLING-INF/content/apps and code associated with login.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Adding logo to Themepage

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Adding logo to Themepage

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Adding logo to Themepage

Commit:
Merge changes from LFS-51 into LFS-34

Commit:
Incomplete work on separating user management into separate module.

Commit:
Separated user management into its own module.

Commit:
Incomplete work on separating user management into separate module.

Commit:
Separated user management into its own module.

zm2he committed:
Experiment for disabling login button on login.

zm2he committed:
Experiment for disabling login button on login.

zm2he committed:
Merge branch 'LFS-34' of https://github.com/ccmbioinfo/lfs into LFS-34

Commit:
Changed typo nulll to null.

Commit:
Changed typo nulll to null.

Commit:
Merge with dev.

Commit:
LFS-34: Add Apache license to main.js of user-viewer module.

Commit:
LFS-34: Add Apache license to main.js of user-viewer module.

Commit:
Merge branch 'dev' into LFS-34 to resolve conflicts.

OrdiNeu committed:
LFS-81 / LFS-83: Pull user data and display on main website
Move Dashboard contents from hard-coded to state-driven

OrdiNeu committed:
LFS-81 / LFS-83: Pull user data and display on main website
Move Dashboard contents from hard-coded to state-driven

OrdiNeu committed:
LFS-81 / LFS-83: Pull user data and display on main website
Move Dashboard contents from hard-coded to state-driven

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove dead html code that affected page formatting, and alter styles
for aesthetics.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove dead html code that affected page formatting, and alter styles
for aesthetics.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove dead html code that affected page formatting, and alter styles
for aesthetics.

OrdiNeu committed:
[misc] Remove obsolete files

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove obsolete files

OrdiNeu committed:
[misc] Remove obsolete files

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Replace the default version of material-ui on the rest of the website
with a downgraded version (3.9.2) as we cannot use the latest version
with material-dashboard-react

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Replace the default version of material-ui on the rest of the website
with a downgraded version (3.9.2) as we cannot use the latest version
with material-dashboard-react

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Replace the default version of material-ui on the rest of the website
with a downgraded version (3.9.2) as we cannot use the latest version
with material-dashboard-react

OrdiNeu committed:
LFS-71: Replace dashboard icon with sign out icon

OrdiNeu committed:
LFS-71: Replace dashboard icon with sign out icon

OrdiNeu committed:
LFS-71: Replace dashboard icon with sign out icon

OrdiNeu committed:
[misc] Move magic number into variable and freeze the version of material-dashboard-react to prevent
crashes

OrdiNeu committed:
[misc] Move magic number into variable and freeze the version of material-dashboard-react to prevent
crashes

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Move magic number into variable and freeze the version of material-dashboard-react to prevent
crashes

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Adding in User Profile route for Bruce

OrdiNeu committed:
[misc] Adding in User Profile route for Bruce

OrdiNeu committed:
[misc] Adding in User Profile route for Bruce

OrdiNeu committed:
[misc] Clean up dead code

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Clean up dead code

OrdiNeu committed:
[misc] Clean up dead code

OrdiNeu committed:
LFS-71: Redirect the user to the login page if they are not logged in

OrdiNeu committed:
LFS-71: Redirect the user to the login page if they are not logged in

OrdiNeu committed:
LFS-71: Redirect the user to the login page if they are not logged in

OrdiNeu committed:
LFS-71: Replace the default web page

OrdiNeu committed:
LFS-71: Replace the default web page

OrdiNeu committed:
LFS-71: Replace the default web page

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Fix double scroll bar on themepage

OrdiNeu committed:
[misc] Fix double scroll bar on themepage

OrdiNeu committed:
[misc] Fix double scroll bar on themepage

OrdiNeu committed:
[misc] Remove old /themepage.html link, fix /content links

OrdiNeu committed:
[misc] Remove old /themepage.html link, fix /content links

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove old /themepage.html link, fix /content links

OrdiNeu committed:
Merge remote-tracking branch 'origin/LFS-94' into LFS-83

OrdiNeu committed:
LFS-71: Poll content/forms/ for data

OrdiNeu committed:
LFS-71: Poll content/forms/ for data

OrdiNeu committed:
LFS-71: Poll content/forms/ for data

Commit:
LFS-34: UI for adding/removing users
Merge with LFS-83 for new UI

Commit:
LFS-34: UI for adding/removing users
Merge with LFS-83 for new UI

OrdiNeu committed:
[misc] Prettify the login page

OrdiNeu committed:
[misc] Prettify the login page

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Prettify the login page

Commit:
LFS-34: New directory for material dashboard react userboard

Commit:
LFS-34: New directory for material dashboard react userboard

Commit:
LFS-34: UI for adding/listing users
Basic listing of users in material dashboard react

Commit:
LFS-34: UI for adding/listing users
Basic listing of users in material dashboard react

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Use /query?query=select%20*%20from%20[lfs:Folder] in order to perform a
JCR-SQL2 query and receive the results back as a .json

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove RTL support

OrdiNeu committed:
[misc] Remove RTL support

OrdiNeu committed:
[misc] Remove RTL support

OrdiNeu committed:
[misc] Further code cleanup

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Further code cleanup

OrdiNeu committed:
[misc] Further code cleanup

OrdiNeu committed:
[misc] Close mini-navbar when the window changes width

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Close mini-navbar when the window changes width

OrdiNeu committed:
[misc] Close mini-navbar when the window changes width

OrdiNeu committed:
[wip] Code documentation and cleanup

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Code documentation and cleanup

OrdiNeu committed:
[wip] Code documentation and cleanup

Commit:
Merge branch 'LFS-83' into LFS-34

zm2he committed:
LFS-34: UI for user management
Resolve sidebar merge conflicts. Set up form for creating new users.

zm2he committed:
LFS-34: UI for user management
Resolve sidebar merge conflicts. Set up form for creating new users.

OrdiNeu committed:
LFS-66: Populate sidebar with all available entries

Commit:
LFS-34: User management page
List groups

Commit:
LFS-34: User management page
List groups

Commit:
LFS-34: User Management Page
Rename files, change layout, add dialog for creating users.

Commit:
LFS-34: User Management Page
Rename files, change layout, add dialog for creating users.

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+        </form>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={() => this.handleClose()}>Close</Button>
```
This line crashes, since this.handleClose() isn't set to anything. Did you mean this.props.handleClose() ?

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+      return response.json();
+    })
+    .then((data) => {
+      console.log(JSON.stringify(data));
```
Remove this console.log and the one on line 151 -- it clutters the log

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+          <CardFooter>
+            <Button onClick={() => this.handleCreateUser()}>Submit</Button>
+          </CardFooter>
+                </Card>  */}        
```
Remove dead code

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboardStyle.jsx:
```diff
@@ -0,0 +1,100 @@
+/*
```
This file is the exact same as dashboardStyle.jsx. Maybe we should combine these into one style .jsx in the themePage/ folder, and just import it from both?

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+
+import  {withStyles} from "@material-ui/core/styles";
+
+//import Button from "material-dashboard-react/dist/components/CustomButtons/Button.js";
```
Remove unused imports, and move imports from the same module to be next to one another.

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+              </CardBody>
+            </Card>
+          </GridItem>
+          
```
There's extra whitespace here and scattered throughout the file.

Commit:
LFS-34: User Management Page
Use Material-Ui input forms

Commit:
LFS-34: User Management Page
Use Material-Ui input forms

OrdiNeu commented on modules/homepage/src/main/frontend/src/themePage/Userboard/userboard.jsx:
```diff
@@ -0,0 +1,281 @@
+    };
+
+  }
+  handleCreateUser() {
```
User creation doesn't appear to work -- I get a 500 (server error) when using it, and it redirects to a 404 (the 404 is probably due to the way the React routing works).

Commit:
LFS-34: User Management Page
Remove create user and group dialogues to separate file.

Commit:
LFS-34: User Management Page
Remove create user and group dialogues to separate file.

OrdiNeu committed:
LFS-66: Populate sidebar with all available entries
Move the data URL from lfs:extensionPointId to lfs:extensionRenderURL

OrdiNeu committed:
LFS-66: Populate sidebar with all available entries
Code cleanup, use the lfs:extensionPointId instead of the
lfs:extensionRenderURL.

Commit:
LFS-34: User Management Page
List users and groups using list with checkbox selection

Commit:
LFS-34: User Management Page
List users and groups using list with checkbox selection

Commit:
LFS-34: User Management Page
Add dialogues for deleting users and groups.

Commit:
LFS-34: User Management Page
Add dialogues for deleting users and groups.

Commit:
LFS-34: User Management Page
Add dialogues for deleting users or groups and changing user passwords. Fix input mismatch for creating new users.

Commit:
LFS-34: User Management Page
Add dialogues for deleting users or groups and changing user passwords. Fix input mismatch for creating new users.

Commit:
LFS-34: User Management Page
Proposed dialogue to add and remove users from groups. Experiments with routing.

Commit:
LFS-34: User Management Page
Proposed dialogue to add and remove users from groups. Experiments with routing.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Move homepage /media to /src/main/media

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Move homepage /media to /src/main/media

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Rename _parseDate to _formateDate

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Rename _parseDate to _formateDate

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Replace lfs:Homepage with lfs:Themepage

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Replace lfs:Homepage with lfs:Themepage

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Fix sidebar home redirect URL

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Fix sidebar home redirect URL

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Add newlines at the end of files

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Add newlines at the end of files

Commit:
LFS-34: User Management Page
Dialogues for adding and removing users from groups completed.

Commit:
LFS-34: User Management Page
Dialogues for adding and removing users from groups completed.

Commit:
Merge branch 'dev' into LFS-34

Commit:
LFS-34: User Management Page
Use principals servlet instead of userManager to send post request for list of users and groups.

Commit:
LFS-34: User Management Page
Use principals servlet instead of userManager to send post request for list of users and groups.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Swap to using curly-brace named imports

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Swap to using curly-brace named imports

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Remove dead code

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Move routes from '/<page>' to '/content.html/<page>' to make the
resultant URLs valid, enabling page refreshes

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Move routes from '/<page>' to '/content.html/<page>' to make the
resultant URLs valid, enabling page refreshes

Commit:
LFS-34: User Management Page
Clean up code and correct error in add/remove users to group which allows single user to be selected and listed multiple times on the interface.

Commit:
LFS-34: User Management Page
Clean up code and correct error in add/remove users to group which allows single user to be selected and listed multiple times on the interface.

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Fix redirect links

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Fix redirect links

OrdiNeu committed:
LFS-81 / LFS-82: Use the material-dashboard-react to theme the website
Remove old homepage code

OrdiNeu committed:
LFS-81 / LFS-83: Use the material-dashboard-react to theme the website
Remove old homepage code

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Switch <div> to <sly> in dataQuery, add newline at end of file

Commit:
LFS-34: User Management Page
Responsive features for displaying users and groups on small screens.

Commit:
LFS-34: User Management Page
Responsive features for displaying users and groups on small screens.

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Use the new adaptTo() function to format Resources to JSON objects

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Comment cleanup

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Use a JsonArrayBuilder instead of manually creating the string

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Switch <div> to <sly>

OrdiNeu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Remove dead code

Commit:
LFS-34: User Management Page
Add searching for users and reorganize table columns.

Commit:
LFS-34: User Management Page
Add searching for users and reorganize table columns.

Commit:
Merge remote-tracking branch 'origin/LFS-83' into LFS-34

OrdiNeu committed:
Merge pull request #30 from ccmbioinfo/LFS-83

 LFS-81 / LFS-83: Use the material-dashboard-react to theme the website

Commit:
LFS-34: User Management Page
Restyle card displaying group information.

Commit:
LFS-34: User Management Page
Restyle card displaying group information.

Commit:
LFS-34: User Management Page
Update material-ui imports to version 4, searching for groups with clear before each search, unsuccessful attempt to resolve React warnings.

Commit:
LFS-34: User Management Page
Update material-ui imports to version 4, searching for groups with clear before each search, unsuccessful attempt to resolve React warnings.

Commit:
LFS-34: User Management Page
Restyle user and group information. Reformat dialogue code and close on submit.

Commit:
LFS-34: User Management Page
Restyle user and group information. Reformat dialogue code and close on submit.

sdumitriu committed:
[misc] Better javadoc and codestyle

sdumitriu committed:
LFS-88 / LFS-90: Build functionality to perform simple queries on imported data
Use the correct Content-Type for the query results

sdumitriu committed:
Merge pull request #38 from ccmbioinfo/LFS-88-standalone

LFS-88 / LFS-90: Build functionality to perform simple queries on imported data

OrdiNeu committed:
LFS-66: Populate sidebar with all available entries
Code cleanup

Commit:
LFS-34: User Management Page
Put dialogs into individual jsx files. Split Userboard component into UsersManager and GroupsManager components for users and groups.

Commit:
LFS-34: User Management Page
Put dialogs into individual jsx files. Split Userboard component into UsersManager and GroupsManager components for users and groups.

Commit:
LFS-34: User Management Page
Fixed bug which forces you to deselect only the user at the top of the list of users to remove each time, instead of letting you deselect any user you choose.

Commit:
LFS-34: User Management Page
Fixed bug which forces you to deselect only the user at the top of the list of users to remove each time, instead of letting you deselect any user you choose.

Commit:
LFS-34: User Management Page
Incomplete pagination for user table.

Commit:
LFS-34: User Management Page
Incomplete pagination for user table.

Commit:
LFS-34: User Management Page
Change typo 'autofocus' to 'autoFocus', remove deadcode.
Functional pagination obtained with following corrections:
Use bind(this) instead of => notation. Stop clearing returnedUserRows and totalUserRows on each load to prevent unwanted reset of page to 0.

Commit:
LFS-34: User Management Page
Change typo 'autofocus' to 'autoFocus', remove deadcode.
Functional pagination obtained with following corrections:
Use bind(this) instead of => notation. Stop clearing returnedUserRows and totalUserRows on each load to prevent unwanted reset of page to 0.

Commit:
LFS-34: User Management Page
Add pagination to unpaginated tables (groups, mobile users).

Commit:
LFS-34: User Management Page
Add pagination to unpaginated tables (groups, mobile users).

Commit:
LFS-34: User Management Page
Remove class for pagination buttons to separate file. Change incorrect named variables in handleChangeRowsPerPage of group manage, and make group manager search function obey pagination.

Commit:
LFS-34: User Management Page
Remove class for pagination buttons to separate file. Change incorrect named variables in handleChangeRowsPerPage of group manage, and make group manager search function obey pagination.

Commit:
LFS-34: User Management Page
Use credentials: 'include' instead of hardcoding and update tables every time a create or delete is used.

Commit:
LFS-34: User Management Page
Use credentials: 'include' instead of hardcoding and update tables every time a create or delete is used.

Commit:
LFS-34: User Management Page
Unfinished attempt to move page into the principals module.

Commit:
LFS-34: User Management Page
Unfinished attempt to move page into the principals module.

OrdiNeu committed:
LFS-66: Populate sidebar with all available entries

Typo fix

Co-Authored-By: zm2he <44327446+zm2he@users.noreply.github.com>

Commit:
LFS-34: User Management Page
Stopgap fix for routes.

Commit:
LFS-34: User Management Page
Fixes after rebasing.

Commit:
Merge branch 'LFS-34' of https://github.com/ccmbioinfo/lfs into LFS-34
